### DISCUSSION TITLE
Rinnovo certificato HAssOS - Caddy 2 - Duckdns

### DISCUSSION THREAD

**Discussion by jraziel09** [indomus/forum#1709] [HUB: Home Assistant]
Ciao a tutti, ho il seguente problema che non riesco a capire:
Da qualche giorno non riesco ad accedere da App puntando su https://xxxxx.duckdns.org:9000/ e neanche dal mio pc o dal browser ma solo una sessione anonima accedo senza problemi.

Ho verificato il certificato di xxxxx.duckdns.org porta 9000 con l’integrazione e mi dice che è scaduto (attributi del sensore Is valid: false ed Error: certificate has expired).

Non riesco a capire quale sia la procedura per rinnovare il certificato in maniera automatica o manuale.

La mia configurazione di HassOS su VM è la seguente:

Porte:
- 9000 su 9000 ip HA
- 443 su 8123 ip HA

configuration.yaml
``` 
http: 
  # PER ACCESSO SENZA CADDY
  #ssl_certificate: /ssl/fullchain.pem 
  #ssl_key: /ssl/privkey.pem
  
  # PER ACCESSO CON CADDY
  use_x_forwarded_for: true
  trusted_proxies: 
    - xxx.xxx.x.xxx/24
``` 
xxx.xxx.x.xxx/24 è l’ip di HA

Caddy 2 configurato come da guida
https://indomus.it/guide/installare-e-configurare-caddy-su-home-assistant-os-hassos-hassio/ 
Caddyfile
```
 (https_header) {
  header {
    Strict-Transport-Security "max-age=31536000; includeSubdomains"
    X-XSS-Protection "1; mode=block"
    X-Content-Type-Options "nosniff"
    X-Frame-Options "SAMEORIGIN"
    Referrer-Policy "same-origin"
  }
}
https://xxxxx.duckdns.org  {
  respond "Saluti dal Crew inDomus"
  file_server
}
https://xxxx.duckdns.org:9000 {
  import https_header
  reverse_proxy http:// xxx.xxx.x.xxx:8123
}

``` 
Registro caddy all’avvio:
non capisco perchè si aspetta un HHTP e non HTTPS

``` 
{"level":"error","ts":1653828248.5106835,"logger":"tls.issuance.acme.acme_client","msg":"challenge failed","identifier":"xxxx.duckdns.org","challenge_type":"tls-alpn-01","problem":{"type":"urn:ietf:params:acme:error:malformed","title":"","detail":"x.xxx.xxx.xxx: Server only speaks HTTP, not TLS","instance":"","subproblems":[]}}
{"level":"error","ts":1653828248.5112972,"logger":"tls.issuance.acme.acme_client","msg":"validating authorization","identifier":"xxxx.duckdns.org","problem":{"type":"urn:ietf:params:acme:error:malformed","title":"","detail":"x.xxx.xxx.xxx: Server only speaks HTTP, not TLS","instance":"","subproblems":[]},"order":"https://acme-v02.api.letsencrypt.org/acme/order/ xxxxx ","attempt":1,"max_attempts":3}
{"level":"error","ts":1653828248.5135245,"logger":"tls.renew","msg":"could not get certificate from issuer","identifier":"xxxx.duckdns.org","issuer":"acme-v02.api.letsencrypt.org-directory","error":"HTTP 400 urn:ietf:params:acme:error:malformed - x.xxx.xxx.xxx: Server only speaks HTTP, not TLS"}
{"level":"warn","ts":1653828248.5163054,"logger":"tls.issuance.zerossl","msg":"missing email address for ZeroSSL; it is strongly recommended to set one for next time"}
{"level":"info","ts":1653828250.566514,"logger":"tls.issuance.zerossl","msg":"generated EAB credentials","key_id":"xxxxx"}
{"level":"warn","ts":1653828273.7805321,"logger":"tls.issuance.acme.acme_client","msg":"HTTP request failed; retrying","url":"https://acme.zerossl.com/v2/DV90/newNonce","error":"performing request: Head \"https://acme.zerossl.com/v2/DV90/newNonce\": http2: timeout awaiting response headers"}
{"level":"info","ts":1653828293.521311,"logger":"tls.issuance.acme","msg":"waiting on internal rate limiter","identifiers":["xxxx.duckdns.org"],"ca":"https://acme.zerossl.com/v2/DV90","account":""}
{"level":"info","ts":1653828293.5214958,"logger":"tls.issuance.acme","msg":"done waiting on internal rate limiter","identifiers":["xxxx.duckdns.org"],"ca":"https://acme.zerossl.com/v2/DV90","account":""}

``` 
Il registro dell’addon duckdns mi da sempre e solo
``` 
[14:34:16] INFO: OK
x.xxx.xxx.xxx
NOCHANGE

``` 

Credo che il problema sia su Caddy e il certificato ma non riesco proprio a capire come risolvere.

**Comment by snakuzzo** [comment]
Il certificato viene rinnovato in automatico...e lo vedi pure dal log di caddy che prova a fare la challenge ma da let's encrypt si becca un http 400.

Ad ogni modo il port forwarding, per come vedo il Caddyfile, deve essere

9000 -> 9000 del server/VM
443 -> 443 del server/VM

questo perché è poi Caddy a preoccuparsi di proxare il tutto verso la 8123 di HA

**Reply by jraziel09** [reply]
Grazie mille Diego, mettendo la 443 su 443 ip HA funziona tutto e il certificato si è aggiornato!
Ho provato 443 su 443 server VM ma non funzionava proprio.
All'inizio usavo la 443 su 8123 per l'accesso senza Caddy e mi ero proprio dimenticato di questo passaggio.

Ma giusto per capire meglio, come mai da sessione anonima funzionava anche senza certificato aggiornato?
